### PR TITLE
Feat/tapc 2054 instance deletion

### PR DESCRIPTION
**Pull request by fredericvilcot**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/pci-instances`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #TAPC-2054
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

## Description

This PR adds a new DELETE action available at instances list.

### CODE REVIEW

**Review by Tsiorifamonjena — APPROVED** [review]

File: packages/manager/apps/pci-instances/src/data/api/instance.ts [hunk 1]
```diff
@@ -27,3 +27,9 @@ export const getInstances = (
       },
     })
     .then((response) => response.data);
+
+export const deleteInstance = (
+  projectId: string,
+  instanceId: string,
+): Promise<null> =>
```
Comment (SimonChaumet) [inline packages/manager/apps/pci-instances/src/data/api/instance.ts:34]: Isn't the returned type void instead ?

File: packages/manager/apps/pci-instances/src/pages/instances/delete/DeleteInstance.page.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+const DeleteInstance: FC = () => {
+  const { t } = useTranslation(['delete', 'common']);
+  const navigate = useNavigate();
+  const { projectId } = useParams() as { projectId: string };
+  const [searchParams] = useSearchParams();
+  const instanceId = searchParams.get('instanceId');
+  const instanceName = searchParams.get('instanceName');
+  const { addError, addSuccess } = useNotifications();
+
+  const handleModalClose = () => {
```
Comment (SimonChaumet) [inline packages/manager/apps/pci-instances/src/pages/instances/delete/DeleteInstance.page.tsx:19]: action methods should be wrapped in useCallback

**Review by ppprevost — APPROVED** [review]

File: packages/manager/apps/pci-instances/src/pages/instances/delete/DeleteInstance.page.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    navigate('..');
+  };
+
+  const handleMutationSuccess = () => {
+    if (instanceId) {
+      updateDeletedInstanceStatus(queryClient, instanceId);
+    }
+    addSuccess(
```
Comment (SimonChaumet) [inline packages/manager/apps/pci-instances/src/pages/instances/delete/DeleteInstance.page.tsx:27]: if we don't have an instanceId we still put a notification ?

File: packages/manager/apps/pci-instances/src/data/api/instance.ts (hunk 1 above)
Comment (fredericvilcot) [reply packages/manager/apps/pci-instances/src/data/api/instance.ts:34]: Nope, API responds null by contract.

File: packages/manager/apps/pci-instances/src/pages/instances/delete/DeleteInstance.page.tsx (hunk 2 above)
Comment (fredericvilcot) [reply packages/manager/apps/pci-instances/src/pages/instances/delete/DeleteInstance.page.tsx:19]: Well, I feel that it's not necessary here as there is not a heavy computation that needs a memoization.

File: packages/manager/apps/pci-instances/src/pages/instances/delete/DeleteInstance.page.tsx (hunk 3 above)
Comment (fredericvilcot) [reply packages/manager/apps/pci-instances/src/pages/instances/delete/DeleteInstance.page.tsx:27]: instancsId is always defined when method is called in render cycle, but yep, I can include it in the if statement 👍

Comment (SimonChaumet) [reply packages/manager/apps/pci-instances/src/pages/instances/delete/DeleteInstance.page.tsx:27]: If the call isn't possible you can add "!" at the end or put it in a casted variable

**Review by SimonChaumet — APPROVED** [review]

File: packages/manager/apps/pci-instances/src/pages/instances/delete/DeleteInstance.page.tsx (hunk 3 above)
Comment (fredericvilcot) [reply packages/manager/apps/pci-instances/src/pages/instances/delete/DeleteInstance.page.tsx:25]: I removed the `if` statement and better handled the `undefined` parameter

**Pull request merged**